### ISSUE TITLE
KafkaProducer - flush 'success' after 30 seconds but no message sent

### ISSUE THREAD

**Issue by kcris** [dpkp/kafka-python#1667]
Hi,
using python 3.7.1, kafka-python 1.4.4, kafka 2.1.0

I forgot to uncomment and set the actual ip for the kafka server in kafka's `config/server.properties` (`listeners` property), which leads to weird results while using kafka-python's producer.

I call `send()` and then `flush(50)`, then `close(50)`
I noticed that it takes 30 seconds (consistently) for `flush` to return, with no error.
But no actual data is received by my broker (which is normal since the listener was incorrectly specified).

What is weird is that I get no exception, no error whatsoever from `flush` or `send`, even if I use a finite timeout. I suspect that 30 seconds is actually some kind of internal timeout (because mine is set to 50).

Bottom line: no data sent, but no error either in `KafkaProducer`'s `send` / `flush`.

Thanks

**Comment by dpkp** [comment]
Errors are returned asynchronously via the future (aka promise) that is
returned from send(). Can you check there?

**Comment by kcris** [comment]
Hi,
yes, I can confirm, checking the future.., I get, as expected, the KafkaTimeoutError.

The question remains though: why don't I get the same exception when doing flush?
Isn't the same semantics for both: wait until the send finishes (with success/failure) and if there's a failure, raise a TimeouException?

according to the docs: no, we don't get the same semantics (my bad for not reading carefully earlier)

https://kafka-python.readthedocs.io/en/master/apidoc/KafkaProducer.html
says that
`flush` (unlike `send`'s future) thinks that if a future is done (either success or failure) - everything is 'ok'

But that is weird....a failed `send` was not actually flush-ed, right? so why ignore the error in `flush`?

Thanks

**Comment by aliartiza75** [comment]
@kcris I am also facing the same issue

**Comment by jeffwidman** [comment]
Some of this behavior was changed in https://github.com/dpkp/kafka-python/pull/1553 to make errors more obvious.
